### ISSUE TITLE
Client never connects if the initial connection attempt fails

### ISSUE THREAD

**Issue by cmotelet** [SignalK/signalk-js-client#90]
In the SignalK server we need an infinite reconnection not only based on a retry number
The code below shows the behavior when connecting to a closed port:
```
wait for incoming client message...
connectOrReconnect call count: 1
receive on error message: connect ECONNREFUSED 127.0.0.1:666
receive on disconnect message
connect promise reject <= the first call to connect return a promise reject
sleep 1 second
connectOrReconnect call count: 2
connect promise resolve <= the second call to connect return a promise resolve
wait for incoming client message...
wait for incoming client message...
<= no further messages received from client
...
```

```
const SignalK = require('@signalk/client')

let count = 1
let client = new SignalK.Client({
  hostname: 'localhost',
  port: 666,
  useTLS: false,
  reconnect: false,
  autoConnect: false,
  maxRetries: 0
})

client.on('connect', () => {
  console.log('receive on connect message')
})

client.on('error', (err) => {
  console.log('receive on error message:', err.message)
})

client.on('disconnect', () => {
  console.log('receive on disconnect message')
})

function connectOrReconnect() {
  console.log('connectOrReconnect call count:', count)
  count +=1
  client.connect().then(
    (resolve) => {
      console.log('connect promise resolve')
    },
    (reject) => {
      console.log('connect promise reject')
      console.log('sleep 1 second')
      setTimeout(connectOrReconnect, 1000);
    }
  )
}

function waitMessage() {
  console.log('wait for incoming client message...')
  setTimeout(waitMessage, 2000);
}
waitMessage()
connectOrReconnect()
```

**Comment by fabdrol** [comment]
@cmotelet is this relevant in conversation https://github.com/SignalK/signalk-js-client/pull/88 and issue https://github.com/SignalK/signalk-js-client/issues/84 ?

**Comment by cmotelet** [comment]
@fabdrol yes #84 and with specific case when you start with closed port

**Comment by fabdrol** [comment]
@cmotelet I just merged #84, is this still an issue with that fixed?

**Comment by fabdrol** [comment]
@cmotelet just merged #84, does that fix this issue or does it need more testing?

**Comment by tkurki** [comment]
This seems to be resolved or needs more details.
